### PR TITLE
spotube: init at 3.4.1

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Related issue: https://github.com/NixOS/nixpkgs/issues/200120

This PR adds 1 package: `spotube`

I used the prebuilt binary distribution, as the `flutter` version of Nixpkgs is not low enough for building from source.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sp/spotube/package.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    changelog = "https://github.com/KRTirtho/spotube/blob/v${finalAttrs.version}/CHANGELOG.md";
+    description = "An open source Spotify client";
+    longDescription = ''
+      Spotube is an open source, cross-platform Spotify client that
+      doesn't require Premium nor uses Electron
+    '';
+    homepage = "https://spotube.netlify.app/";
+    license = lib.licenses.bsdOriginal;
+    maintainers = with lib.maintainers; [ tomasajt ];
+    platforms = [ "x86_64-linux" ];
```
Comment (storopoli) [inline pkgs/by-name/sp/spotube/package.nix:72]: What happens if you naively add darwin Apple Silicon and Intel here?

Comment (storopoli) [reply pkgs/by-name/sp/spotube/package.nix:72]: Thanks for this :)

**Comment by Steinhagen** [comment]
Gave it a try again and everything is working as expected. Looks good!

**Comment by nephalemsec** [comment]
I've made some amendments so that the darwin and linux derivations are in a single file:

```nix
{ stdenv, fetchurl, lib, makeDesktopItem, makeWrapper, undmg, autoPatchelfHook
, dpkg, wrapGAppsHook

, libappindicator, libnotify, libsecret, mpv-unwrapped, xdg-user-dirs }:
let
  pname = "spotube";
  version = "3.4.0";
  appname = "Spotube";
  meta = with lib; {
    description = ''
      An open source, cross-platform Spotify client compatible across multiple platforms
              utilizing Spotify's data API and YouTube (or Piped.video or JioSaavn) as an audio source,
              eliminating the need for Spotify Premium'';
    homepage = "https://spotube.netlify.app/";
    downloadPage = "https://github.com/KRTirtho/spotube/releases";
    license = licenses.bsdOriginal;
    maintainers = with lib.maintainers; [ tomasajt ];
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
  };

  filename = if stdenv.isDarwin then
    "${appname}-macos-universal.dmg"
  else
    "${appname}-linux-x86_64.deb";

  src = fetchurl {
    url =
      "https://github.com/KRTirtho/spotube/releases/download/v${version}/${filename}";
    hash = if stdenv.isDarwin then
      "sha256-dif2ec37QL9FVPGgqqvtB2zRX6P3NCth5FFnpASR/As="
    else
      "sha256-XBciTNWsmaY1C8ji+SCWWP/M/uIDZHM9fi9wucHJsBY=";
  };

  icon = fetchurl {
    url =
      "https://raw.githubusercontent.com/KRTirtho/spotube/nightly/assets/spotube-logo.png";
    hash = "sha256-gyzP3ynzuZFN0osmEAjwGfYmbfVxE9Nm2fLJ8q8G6BE=";
  };

  desktopItem = makeDesktopItem {
    name = "${pname}";
    desktopName = "${appname}";
    comment = "An open source spotify client.";
    icon = "${pname}";
    exec = "feishin %U";
    categories = [ "Audio" ];
    mimeTypes = [ "x-scheme-handler/${pname}" ];
  };

  darwin = stdenv.mkDerivation {
    inherit pname version src appname;

    meta = meta // { platforms = [ "x86_64-darwin" "aarch64-darwin" ]; };

    sourceRoot = "${appname}.app";
    nativeBuildInputs = [ makeWrapper undmg ];
    installPhase = ''
      runHook preInstall
      mkdir -p $out/{Applications/${appname}.app,bin}
      cp -R . $out/Applications/${appname}.app
      makeWrapper $out/Applications/${appname}.app/Contents/MacOS/${appname} $out/bin/${pname}
      runHook postInstall
    '';
  };

  linux = stdenv.mkDerivation {
    inherit pname version src appname;

    meta = meta // { platforms = [ "x86_64-linux" ]; };

    nativeBuildInputs = [ autoPatchelfHook dpkg makeWrapper wrapGAppsHook ];

    buildInputs = [ libappindicator libnotify libsecret mpv-unwrapped ];

    dontWrapGApps = true;

    installPhase = ''
      runHook preInstall
      mkdir -p $out
      mv usr/* $out
      runHook postInstall
    '';

    preFixup = ''
      patchelf $out/share/spotube/lib/libmedia_kit_native_event_loop.so \
          --replace-needed libmpv.so.1 libmpv.so.2
    '';

    postFixup = ''
      makeWrapper $out/share/spotube/spotube $out/bin/spotube \
          "''${gappsWrapperArgs[@]}" \
          --prefix LD_LIBRARY_PATH : $out/share/spotube/lib:${
            lib.makeLibraryPath [ mpv-unwrapped ]
          } \
          --prefix PATH : ${lib.makeBinPath [ xdg-user-dirs ]}
    '';
  };

in if stdenv.isDarwin then darwin else linux

```

I'm not on my linux boxen atm so I'm assuming your above derivation builds correctly and just modularised it. Appreciate the effort you put in <3

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
Overall looks good, but 3.4.1 just got released so we might want to update to that before we merge this PR.

**Pull request merged**